R/defineModel.R

```diff
@@ -1,18 +1,126 @@
-                        progress = FALSE, Msteps = NULL, increment.factor=1 , fac.oldxsi=0, export = list(logfile = TRUE, systemfile = FALSE, history = TRUE, covariance = TRUE, reg_coefficients = TRUE, designmatrix = FALSE) )   {
+defineModel <- function(dat, items, id, splittedModels = NULL,
+                        irtmodel = c("1PL", "2PL", "PCM", "PCM2", "RSM", "GPCM", "2PL.groups", "GPCM.design", "3PL"),
+                        qMatrix=NULL, DIF.var=NULL, HG.var=NULL, group.var=NULL,
+                        weight.var=NULL, anchor = NULL, domainCol=NULL, itemCol=NULL,
+                        valueCol=NULL,check.for.linking = TRUE, minNperItem = 50,
+                        removeMinNperItem = FALSE, boundary = 6, remove.boundary = FALSE,
+                        remove.no.answers = TRUE, remove.no.answersHG = TRUE,
+                        remove.missing.items = TRUE, remove.constant.items = TRUE,
+                        remove.failures = FALSE, remove.vars.DIF.missing = TRUE,
+                        remove.vars.DIF.constant = TRUE, verbose=TRUE, software = c("conquest","tam"),
+                        dir = NULL, analysis.name, schooltype.var = NULL, model.statement = "item",
+                        compute.fit = TRUE, pvMethod = c("regular", "bayesian"),
+                        fitTamMmlForBayesian = TRUE, n.plausible=5, seed = NULL,
+                        conquest.folder=NULL, constraints=c("cases","none","items"),
+                        std.err=c("quick","full","none"), distribution=c("normal","discrete"),
+                        method=c("gauss", "quadrature", "montecarlo", "quasiMontecarlo"),
+                        n.iterations=2000, nodes=NULL, p.nodes=2000, f.nodes=2000,
+                        converge=0.001, deviancechange=0.0001, equivalence.table=c("wle","mle","NULL"),
+                        use.letters=FALSE, allowAllScoresEverywhere = TRUE, guessMat = NULL,
+                        est.slopegroups = NULL, fixSlopeMat = NULL, slopeMatDomainCol=NULL,
+                        slopeMatItemCol=NULL, slopeMatValueCol=NULL, progress = FALSE,
+                        Msteps = NULL, increment.factor=1 , fac.oldxsi=0,
+                        export = list(logfile = TRUE, systemfile = FALSE, history = TRUE,
+                                      covariance = TRUE, reg_coefficients = TRUE, designmatrix = FALSE) ){
+
+### checking/asserting the arguments -------------------------------------------
+  # dat
+  ismc <- attr(dat, "multicore")
+  dat  <- eatTools::makeDataFrame(dat, name = "dat")
+
+  # data frame: qMatrix
+  checkmate::assert_data_frame(qMatrix, null.ok = TRUE, col.names = "named", min.cols = 2)
+  warning(paste0("The first column of the data frame `qMatrix` should be called `item`, but is called ",
+                 colnames(qMatrix)[1], "."))
+
+  # dataframe: anchor
```

Comment:
checks for `anchor` argument are very useful, but it's the same story as with the Q matrix: if the user defines an `anchor` argument, it is directly passed to the `anker()` function. (Inconsistently, the argument in the `anker()` function is called `prm`.) 
I would suggest to execute this assertions/checks within the `anker()` function.

I am aware that this differs somewhat from what I used to say (“Everyone checks as high up as possible and only in user-level functions!”) ... Well, what do I care what I said yesterday? 
But I think it would be clearer this way. For example, if you ever ask yourself in the future: Do we have yet implemented assertions for the anchor frame? Then you know immediately where to look for it, namely in the function that processes the anchor parameter if the user has specified one.